fem/pgridfunc.cpp

```diff
@@ -908,6 +908,100 @@ void ParGridFunction::SaveAsOne(const char *fname, int precision) const
    SaveAsOne(ofs);
 }
 
+void ParGridFunction::SaveAsSerial(const char *fname, int precision,
+                                   int save_rank) const
+{
+   GridFunction serialgf = GetSerialGridFunction(save_rank);
+   ParMesh *pmesh = ParFESpace()->GetParMesh();
+   if (pmesh->GetMyRank() == save_rank)
+   {
+      serialgf.Save(fname, precision);
+   }
+   MPI_Barrier(pmesh->GetComm());
+}
+
+GridFunction ParGridFunction::GetSerialGridFunction(int save_rank) const
+{
+   ParFiniteElementSpace *pfespace = ParFESpace();
+   ParMesh *pmesh = pfespace->GetParMesh();
+
+   Mesh serialmesh = pmesh->GetSerialMesh(save_rank);
+
+   int vdim = pfespace->GetVDim();
+   auto *fespace_serial = new FiniteElementSpace(&serialmesh,
+                                                 pfespace->FEColl(),
+                                                 vdim,
+                                                 pfespace->GetOrdering());
+
+   GridFunction gf_serial(fespace_serial);
+   Array<double> vals;
+   Array<int> dofs;
+   MPI_Status status;
+   int n_send_recv;
+
+   int MyRank = pmesh->GetMyRank(),
+       NRanks = pmesh->GetNRanks();
+   MPI_Comm MyComm = pmesh->GetComm();
```

Comment:
There's a mixture of camel case, snake case, and pascal case throughout this PR... maybe worth making it more consistent but not a big deal.

Reply:
resolved in https://github.com/mfem/mfem/pull/3588/commits/0bf3e290e043cdce8cbdb358250b044c87c7d02d, thanks!